test/blackbox-tests/test-cases/pkg/build-pkg-deps.t

```diff
@@ -0,0 +1,74 @@
+This test verifies the @pkg-deps alias fetch and build the project dependencies
+without building the project itself.
+
+  $ . ./helpers.sh
+
+Create a fake library to install in the target project as a dependency:
+  $ mkdir foo
+  $ cd foo
+  $ cat > dune-project <<EOF
+  > (lang dune 3.16)
+  > (package (name foo))
+  > EOF
+  $ cat > foo.ml <<EOF
+  > let foo = "Hello, World!"
+  > EOF
+  $ cat > dune <<EOF
+  > (library
+  >  (public_name foo))
+  > EOF
+  $ cd ..
+  $ tar cf foo.tar foo
+  $ rm -rf foo
+
+Make an opam package for the library:
+  $ mkpkg foo <<EOF
+  > build: [
+  >   [
+  >     "dune"
+  >     "build"
+  >     "-p"
+  >     name
+  >     "@install"
+  >   ]
+  > ]
+  > url {
+  >  src: "$PWD/foo.tar"
+  > }
+  > EOF
+
+Create a project using the fake library as a dependency:
+  $ cat > dune-project << EOF
+  > (lang dune 3.16)
+  > (package
+  >  (name bar)
+  >  (allow_empty)
+  >  (depends foo))
+  > EOF
+  $ add_mock_repo_if_needed
+  $ dune pkg lock
```

Comment:
Why do you need to involve the solver in this test?

Reply:
It was the easiest way to write the test. However, it should be possible to write it without using the `lock` command. I'll rework it.